### PR TITLE
Enhanced the xacml policy loading flow from filesystem preserving the default flow

### PR DESCRIPTION
**Pull request by ashendes**
**Fixes https://github.com/wso2/product-is/issues/9207**

Enhance the XACML policy loading flow from the filesystem, preserving the default flow. 75% improvement in the policy loading time is observed.

This enhanced loading flow will be used when starting the server with the following property.

```
./wso2server.sh -DenableEnhancedXACMLLoading=true
````

or by adding it to the wso2server.sh file.

### CODE REVIEW

**Review by vihanga-liyanage — APPROVED** [review]

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java [hunk 1]
```diff
@@ -307,6 +317,36 @@ protected void activate(ComponentContext ctxt) {
...
     }
 
+    /**
+     * Adds policy files with unique policyIDs to the registry.
+     *
+     * @param policyIdList
+     * @param fileList
+     * @return
+     * @throws IOException
+     */
```
Comment (sachiniWettasinghe) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java:327]: Complete the javadoc with descriptions

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java [hunk 2]
```diff
@@ -307,6 +317,36 @@ protected void activate(ComponentContext ctxt) {
...
+        for (File policyFile : fileList) {
+            if (policyFile.isFile()) {
+                PolicyDTO policyDTO = new PolicyDTO();
+                policyDTO.setPolicy(FileUtils.readFileToString(policyFile));
+                if (!policyIdList.contains(policyDTO.getPolicyId())) {
+                    try {
+                        EntitlementUtil.addFilesystemPolicy(policyDTO, registryService
+                                .getGovernanceSystemRegistry(), true);
+                    } catch (Exception e) {
+                        // log and ignore
```
Comment (sachiniWettasinghe) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java:340]: Format the comment by starting with a capital letter and ending with a '.'. And add a meaningful sentence here.

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java (hunk 1 above)
Comment (ashendes) [reply components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java:327]: Fixed in https://github.com/wso2/carbon-identity-framework/pull/3110

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java (hunk 2 above)
Comment (ashendes) [reply components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java:340]: Fixed in https://github.com/wso2/carbon-identity-framework/pull/3110

**Pull request merged**